lxd/resource_lxd_container.go

```diff
@@ -309,6 +309,9 @@ func resourceLxdContainerExists(d *schema.ResourceData, meta interface{}) (exist
 	exists = false
 
 	ct, err := client.ContainerState(name)
+	if err != nil && err.Error() == "not found" {
+		err = nil
```

Comment:
The usual terraform behaviour is to clear the resource ID if we detect the resource no longer exists, for example see the Google Compute Instance code: 
https://github.com/hashicorp/terraform/blob/master/builtin/providers/google/resource_compute_instance.go#L334

We should do the same here, otherwise as @jtopjian points out, Terraform will continue to believe the resource exists.